src/named_tuple.cr

```diff
@@ -122,16 +153,51 @@ struct NamedTuple
     fetch(key) { raise KeyError.new "Missing named tuple key: #{key.inspect}" }
   end
 
-  # Returns the value for the given *key*, if there's such key, otherwise returns `nil`.
+  # Returns the value type for the given *key* if there is such a key, otherwise
+  # raises `KeyError`.
+  # Read the type docs to understand the difference between indexing with a
+  # literal or a variable.
+  #
+  # ```
+  # alias Foo = NamedTuple(name: String, year: Int32)
+  #
+  # Foo[:name]       # => String
+  # Foo["year"]      # => Int32
+  # Foo["year"].zero # => 0
+  # Foo[:other]      # => Error: missing key 'other' for named tuple NamedTuple(name: String, year: Int32).class
+  #
+  # key = :year
+  # Foo[key]      # => Int32
+  # Foo[key].zero # Error: undefined method '[]' for NamedTuple(name: String, year: Int32).class
```

Comment:
Shouldn't it be:
```suggestion
  # Foo[key].zero # Error: undefined method 'zero' for String.class (compile-time type is (Int32.class | String.class | Nil))
```

Reply:
it's not nilable (`self.[]?` is), other than that yes

Reply:
Wrong copy-paste, sorry!